### PR TITLE
[Split Build] Refactor split build binary builds into their own workflows and move split build binary builds to periodic

### PR DESCRIPTION
**Pull request by PaliC**
As we need to move split build binary tests from trunk to periodic this pr, refactors those jobs out into its own workflow to achieve this.

cc @albanD

### CODE REVIEW

File: .github/scripts/generate_binary_build_matrix.py [hunk 1]
```diff
@@ -400,7 +403,44 @@ def generate_wheels_matrix(
                         ),
                     }
                 )
-                if arch_version != "cuda-aarch64":
+            elif use_split_build and arch_version in ["cpu"] and os == "linux":
```
Comment (malfet) [inline .github/scripts/generate_binary_build_matrix.py:406]: Nit
```suggestion
            elif use_split_build and f"{os}-{arch_version}" == "linux-cpu":
```

File: .github/scripts/generate_binary_build_matrix.py [hunk 2]
```diff
@@ -325,6 +325,7 @@ def generate_wheels_matrix(
     os: str,
     arches: Optional[List[str]] = None,
     python_versions: Optional[List[str]] = None,
+    use_split_build: Optional[bool] = False,
```
Comment (malfet) [inline .github/scripts/generate_binary_build_matrix.py:328]: Why is it optional? How `use_split_build is None` are semantically different than `use_split_build = False`?
```suggestion
    use_split_build: bool = False,
```

File: .github/scripts/generate_binary_build_matrix.py [hunk 3]
```diff
@@ -385,6 +387,7 @@ def generate_wheels_matrix(
                         "desired_cuda": translate_desired_cuda(
                             gpu_arch_type, gpu_arch_version
                         ),
+                        "use_split_build": "True",
```
Comment (malfet) [inline .github/scripts/generate_binary_build_matrix.py:390]: Or if you prefer to add it only if true, you can use something like `{**{original_dict_defines}, **({"use_split_build": "True"} if use_split_build else {})}`
```suggestion
                        "use_split_build": "True" if use_split_build else "False",
```

Comment (PaliC) [reply .github/scripts/generate_binary_build_matrix.py:390]: I think having it there explicitly is reasonable especially if we make it the default.

**Review by malfet — APPROVED** [review]
LGTM, thank you for the updates.

**Pull request closed without merging**